### PR TITLE
Internal: Nested Infra - Fix Experiments manager module dependency [ED-6757]

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

File: core/experiments/manager.php [hunk 1]
```diff
@@ -103,10 +103,12 @@ public function add_feature( array $options ) {
...
-				if ( ! class_exists( $dependency ) ) {
-					throw new \Exception( sprintf( 'Dependency class %s not found', $dependency ) );
+				$feature = $this->get_features( $dependency );
+				if ( ! class_exists( $dependency ) && ! empty( $feature ) ) {
+					$experimental_data['dependencies'][ $key ] = new Wrap_Core_Dependency( $feature );
+				} else {
+					$experimental_data['dependencies'][ $key ] = $dependency::instance();
 				}
-
 				$experimental_data['dependencies'][ $key ] = $dependency::instance();
```
Comment (iNewLegend) [inline core/experiments/manager.php:112]:
```suggestion
```

File: core/experiments/wrap-core-dependency.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	public function __construct( $feature_data ) {
+		$this->feature_data = $feature_data;
+	}
+
+	public function get_name() {
+		return $this->feature_data['name'];
+	}
+
+	public static function instance( $feature_data ) {
+		return new self( $feature_data );
```
Comment (matipojo) [inline core/experiments/wrap-core-dependency.php:21]:
```suggestion
		return new static( $feature_data );
```

**Review by matipojo — CHANGES_REQUESTED** [review]
Please cover with tests

File: tests/phpunit/elementor/core/experiments/test-manager.php [hunk 3]
```diff
@@ -53,6 +54,33 @@ public function test_add_feature() {
...
+		];
+
+		$this->add_test_feature( $test_core_feature );
+
+		// Act.
+		$depended_feature = $this->add_test_feature( $depended_feature );
+
+		$depended_feature_dependency = $depended_feature['dependencies'][0];
+		// Assert.
+
```
Comment (matipojo) [inline tests/phpunit/elementor/core/experiments/test-manager.php:79]:
```suggestion

		// Assert.
```

**Review by matipojo — CHANGES_REQUESTED** [review]

**Pull request merged**